### ISSUE TITLE
Quickstart guide error: wait() from trait can only be used if the trait is in scope

### ISSUE THREAD

**Issue by ZelphirKaltstahl** [actix/actix-web#135]
I am looking at the quickstart guide at <https://actix.github.io/actix-web/guide/qs_3_5.html>. There at the example code for spawning another actix system using a `Thread` with `spawn()`.

I have the following code:

``` rust
// actix-web the server framework we are using.
extern crate actix;
extern crate actix_web;
use actix::*;
use actix_web::*;

use std::thread;
use std::sync::mpsc;

mod model;
use model::*;

// Globals are declared outside all other scopes.
static PORT: u32 = 8000;

// A request handler is a function that accepts a HttpRequest instance as its only parameter
// and returns a type that can be converted into HttpResponse:
fn index(_req: HttpRequest) -> String {
    String::from("Hello World!")
}


fn main() {
    let (tx, rx) = mpsc::channel();

    // Start stuff in a new thread.
    thread::spawn(move || {
        // HttpServer is an actix actor.
        // It has to be initialized within properly configured actix system.
        // Here we initialize such a system.
        let sys = actix::System::new("rugi");

        println!("Trying to start on localhost:{}.", PORT.to_string());
        // run a new server on some port
        let addr = HttpServer::new(
            // make a new application
            // apparently here a lambda expression is needed
            // This is what the guide calls "Application factory",
            // because you get a new Application object when you call this procedure.
            || Application::new()
                .prefix("/")
                .resource("/index", |req| req.f(index)))
            .bind(format!("localhost:{}", PORT)).expect(&format!("Could not bind to port {}", PORT))
            .shutdown_timeout(60)
            .start();  // <- use start_tls() for TLS

        let _ = tx.send(addr);
        let _ = sys.run();
    });

    // Receive the address of the started server from the channel.
    let addr = rx.recv().unwrap();
    // Send a stop message to the server's address.
    let _ = addr.send(server::StopServer{graceful:true}).wait();
}
```

Everything seems to work, except that `wait` is apparently not working:

```
items from traits can only be used if the trait is in scope [E0599]

the method `wait` exists but the following trait bounds were not satisfied:
`actix::dev::Request<actix::Syn, actix_web::HttpServer<actix_web::application::HttpApplication>, actix_web::server::StopServer> : actix::ContextFutureSpawner<_>`
`&actix::dev::Request<actix::Syn, actix_web::HttpServer<actix_web::application::HttpApplication>, actix_web::server::StopServer> : actix::ContextFutureSpawner<_>`
`&mut actix::dev::Request<actix::Syn, actix_web::HttpServer<actix_web::application::HttpApplication>, actix_web::server::StopServer> : actix::ContextFutureSpawner<_>` [E0599]

no method named `wait` found for type `actix::dev::Request<actix::Syn, actix_web::HttpServer<actix_web::application::HttpApplication>, actix_web::server::StopServer>` in the current scope [E0599]
```

Now I am unsure how to go on.

**Comment by fafhrd91** [comment]
`add.send()` returns `Future`, you should be able to use `wait` method.
try to add `use futures::Future;` at the top of the file.

**Comment by ZelphirKaltstahl** [comment]
Adding `use futures::Future;` leads to:

```
unresolved import `futures` (Maybe a missing `extern crate futures;`?) [E0432]
```

It seems I did not have that crate. I went to crates.io, found it there and added `futures = "^0.1.19"` to my TOML file. Now the whole issue is gone, so it was the missing dependency.

My suggestion is to add that somewhere in the quickstart (if I did not overlook it) or maybe even make it a dependency of actix, if it is that much required.

Thanks for helping out!
